### PR TITLE
Error Classes

### PR DESCRIPTION
**Pull request by akapur99**
Addresses: https://app.zenhub.com/workspaces/mojaloop-project-59edee71d1407922110cf083/issues/mojaloop/project/1688

Create and implement specialized error classes

### CODE REVIEW

File: src/models/errors.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ --------------
+ ******/
+
+import { Consent } from './consent'
+
+export class MissingConsentFieldsError extends Error {
+  public consent: Consent
+
+  public constructor(consent: Consent) {
+    super(' Necessary Consent fields are missing')
```
Comment (kkzeng) [inline src/models/errors.ts:36]: nit: unnecessary whitespace in the string

**Review by kkzeng — APPROVED** [review]
Lgtm - just a minor nit I found

File: src/models/errors.ts (hunk 1 above)
Comment (akapur99) [inline src/models/errors.ts:36]:
```suggestion
    super('Necessary Consent fields are missing')
```
